app/javascript/components/namespace-form/index.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { API } from '../../http_api';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import schema from './namespace-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import handleFailure from '../../helpers/handle-failure';
+
+const getInitialValues = (isDomain, namespaceId, f) => {
+  miqSparkleOn();
+  API.get(`/api/automate_${isDomain ? 'domains' : 'namespaces'}/${namespaceId}?attributes=name,description,enabled,source`).then((result) => {
+    miqSparkleOff();
+    f.setInitialValues({ name: result.name, description: result.description, enabled: result.enabled });
+    // FIXME Retrieve readability from the API directly instead of the source (property is missing from the API)
+    f.setDisabled(!['user', 'user_locked'].includes(result.source));
+  });
+};
+
+const handleSubmit = (isDomain, namespaceId, parentId, values, redirectUrl) => {
+  miqSparkleOn(); // miqRedirectBack turns it off
+  const action = namespaceId ? 'edit' : 'create';
+  const flashMessage = __(`Automate ${isDomain ? 'Domain' : 'Namespace'} "${values.name}" was ${namespaceId ? 'saved' : 'added'}.`);
+
+  // The same component is being used for ae_domains and ae_namespaces
+  API.post(`/api/automate_${isDomain ? 'domains' : 'namespaces'}/${namespaceId || ''}`, {
+    action, parent_id: parentId, name: values.name, description: values.description, enabled: values.enabled,
+  }, { skipErrors: [400] })
+    .then(() => miqRedirectBack(flashMessage, 'success', redirectUrl))
+    .catch(handleFailure);
+};
+
+const NamespaceForm = (props) => {
+  const { isDomain, namespaceId, parentId } = props;
+  const [initialValues, setInitialValues] = useState({ name: '', description: '', enabled: true });
+  const [disabled, setDisabled] = useState(false);
+  const [loaded, setLoaded] = useState(false);
+  const redirectUrl = '/miq_ae_class/explorer';
+  // eslint-disable-next-line max-len
+  const cancelMessage = __(`${initialValues.name ? 'Edit' : 'Add'} of ${isDomain ? 'Domain' : 'Namespace'} "${initialValues.name || ''}" was cancelled by user.`);
```

Comment:
You can take all conditions outside and make the line shorter.

Also, this will not work, for dynamic strings use `sprintf` (you will find examples in different forms)